### PR TITLE
[Fix #526] Document and give example of JSON in data

### LINKED ISSUE #526: Document and give example of JSON in data
Document and give example of how a JSON string shall be formatted in `data` field when using data in context and jq expression

```
transition:
      nextState: WaitForSaveTransformationCompletionEvent
      produceEvents:
        - eventRef: saveTransformationEvent
          data: '{"gitRepo":".repositoryURL", "branch": ".targetBranch", "token": ".token", "workspaceId": ".workspaceId", "projectId": ".projectId", "transformId": ".transformId ", "workflowCallerId": "$WORKFLOW.instanceId"}'
```
 was not working as the `data` was not parsed as expected when publishing event:
```
ProcessInstanceDataEvent {specVersion=1.0, id='338d205f-c266-42c5-9187-bd988dd1f756', source=/process/m2k, type='save-transformation', time=2024-01-30T08:48:05.895695548Z, subject='null', dataContentType='null', dataSchema=null, data={"gitRepo":".repositoryURL","branch":".targetBranch","token":".token","workspaceId":".workspaceId","projectId":".projectId","transformId":".transformId ","workflowCallerId":"$WORKFLOW.instanceId"}, kogitoProcessInstanceId='6311bf4e-c490-4473-8e22-bba027730530', kogitoRootProcessInstanceId='null', kogitoProcessId='m2k', kogitoRootProcessId='null', kogitoAddons='null', kogitoIdentity='null', extensionAttributes={kogitoproctype=SW, kogitoprocinstanceid=6311bf4e-c490-4473-8e22-bba027730530, kogitoprocist=Active, kogitoprocversion=1.0, kogitoprocid=m2k}}
```

Instead the `data` string shall be:
```
data: "{gitRepo:.repositoryURL|sub(\"http(s)?://\";\"ssh://\"), branch: .targetBranch, token: .token, workspaceId: .workspaceId, projectId: .projectId, transformId: .transformId, workflowCallerId: $WORKFLOW.instanceId}"
``` 
When published:
```
ProcessInstanceDataEvent {specVersion=1.0, id='293b7596-1253-4c98-ac12-4e9825f01bce', source=/process/m2k, type='save-transformation', time=2024-01-30T13:24:14.595998574Z, subject='null', dataContentType='null', dataSchema=null, data={"gitRepo":"ssh://bitbucket.org/gfarache31/m2k-test","branch":"aaaaaaasssss","token":null,"workspaceId":"b93980cb-3943-4223-9441-8694c098eeb9","projectId":"9b305fe3-d441-48ce-b01b-d314e86e14ec","transformId":"723dce89-c25c-4c7b-9ef3-842de92e6fe6","workflowCallerId":"7ddb5193-bedc-4942-a857-596b31f377ed"}, kogitoProcessInstanceId='7ddb5193-bedc-4942-a857-596b31f377ed', kogitoRootProcessInstanceId='null', kogitoProcessId='m2k', kogitoRootProcessId='null', kogitoAddons='null', kogitoIdentity='null', extensionAttributes={kogitoproctype=SW, kogitoprocinstanceid=7ddb5193-bedc-4942-a857-596b31f377ed, kogitoprocist=Active, kogitoprocversion=1.0, kogitoprocid=m2k}}
```

### PR DESCRIPTION
**Pull request by fjtirado**
Fixes https://github.com/apache/incubator-kie-kogito-docs/issues/526

**Description:**
As per subject. Also adds function of type expression and more info on $SECRET data type.

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc [hunk 1]
```diff
@@ -13,10 +13,27 @@ Each workflow instance is associated with a data model. A data model consists of
 
 The workflow expressions in the link:{spec_doc_url}#workflow-expressions[Serverless Workflow specification] are used to interact with the data model. The supported expression languages include link:{jsonpath_url}[JsonPath] and link:{jq_url}[jq]. jq expression language is the default language. However, you can change the expression language to JsonPath using the `expressionLang` property. 
 
-This document describes the usage of jq expressions in switch state conditions, action function arguments, and data filtering. 
+This document describes the usage of jq expressions in funtions, switch state conditions, action function arguments, data filtering and event publishing.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc:16]:
```suggestion
This document describes the usage of jq expressions in functions, switch state conditions, action function arguments, data filtering, and event publishing. 
```

File: serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc [hunk 2]
```diff
@@ -13,10 +13,27 @@ Each workflow instance is associated with a data model. A data model consists of
...
 
-This document describes the usage of jq expressions in switch state conditions, action function arguments, and data filtering. 
+This document describes the usage of jq expressions in funtions, switch state conditions, action function arguments, data filtering and event publishing. 
 
 JQ expression might be tricky to master, for non trivial cases, it is recommended to use helper tools like link:{jq_play}[JQ Play] to validate the expression before including it in the workflow file. 
 
+[[ref-example-jq-expression-function]]
+== Example  of jq expression in functions
+
+Expressions can be used in functions of type `expression` to manipulate the workflow model. As any other function the result of the expression evaluation will be merged into the model
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc:23]:
```suggestion
Expressions can be used in functions of type `expression` to manipulate the workflow model. As with any other function, the result of the expression evaluation will be merged into the model
```

File: serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc [hunk 3]
```diff
@@ -197,13 +214,37 @@ You can find an example of event data filtering in the link:{kogito_sw_examples_
 ----
 
 The previous example of the event filter copies the content of CloudEvent data `result` field into the workflow model `move` field.
+
+[[ref-example-jq-expression-event-publishing]]
+== Example of jq expressions in event publishing. 
+
+When publishing a Cloud Event, you can select the data that is being published using a jq expression that generates a json object. Note that in yaml double quotes are required to allow using `{}` characters.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc:221]:
```suggestion
When publishing a Cloud Event, you can select the data that is being published using a jq expression that generates a JSON object. Note that in yaml double quotes are required to allow using `{}` characters. 
```

File: serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc [hunk 4]
```diff
@@ -197,13 +214,37 @@ You can find an example of event data filtering in the link:{kogito_sw_examples_
...
+[source,yaml]
+----
+transition:
+      nextState: WaitForSaveTransformationCompletionEvent
+      produceEvents:
+        - eventRef: saveTransformationEvent
+          data: "{gitRepo:.repositoryURL|sub(\"http(s)?://\";\"ssh://\"), branch: .targetBranch, token: .token, workspaceId: .workspaceId, projectId: .projectId, transformId: .transformId, workflowCallerId: $WORKFLOW.instanceId}" 
+----
+
+In previous example, a Cloud Event is published when the state transitioned. Its `data` field looks like
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc:233]:
```suggestion
In the previous example, a CloudEvent was published when the state transitioned. Its `data` field looks like 
```

File: serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc [hunk 5]
```diff
@@ -212,6 +253,15 @@ So, assuming you have added to your `application.properties` a line with the `my
 }
 ----
 
+`$SECRET` always return a value of type string. If you want to use it into a comparison with a value that is not a string, you need to perform the conversion explicitly. In next example, `retries` property is a number, so we need to convert the property value accordingly by calling `tonumber` builtin jq function.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc:256]:
```suggestion
`$SECRET` always returns a value of type string. If you want to use it in a comparison with a value that is not a string, you need to perform the conversion explicitly. In the next example, the `retries` property is a number, so we need to convert the property value accordingly by calling the `tonumber` built-in jq function. 
```

**Review by ricardozanini — APPROVED** [review]

**Review by domhanak — APPROVED** [review]

**Comment by ricardozanini** [comment]
@kaldesai can you please take a look?

File: serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc [hunk 6]
```diff
@@ -13,10 +13,27 @@ Each workflow instance is associated with a data model. A data model consists of
...
 
-This document describes the usage of jq expressions in switch state conditions, action function arguments, and data filtering. 
+This document describes the usage of jq expressions in functions, switch state conditions, action function arguments, data filtering, and event publishing. 
 
 JQ expression might be tricky to master, for non trivial cases, it is recommended to use helper tools like link:{jq_play}[JQ Play] to validate the expression before including it in the workflow file. 
 
+[[ref-example-jq-expression-function]]
+== Example  of jq expression in functions
+
+Expressions can be used in functions of type `expression` to manipulate the workflow model. As with any other function, the result of the expression evaluation will be merged into the model
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc:23]:
```suggestion
Expressions can be used in functions of type `expression` to manipulate the workflow model. As with any other function, the result of the expression evaluation will be merged into the model.
```

File: serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc [hunk 7]
```diff
@@ -212,6 +253,15 @@ So, assuming you have added to your `application.properties` a line with the `my
 }
 ----
 
+`$SECRET` always returns a value of type string. If you want to use it in a comparison with a value that is not a string, you need to perform the conversion explicitly. In the next example, the `retries` property is a number, so we need to convert the property value accordingly by calling the `tonumber` built-in jq function.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc:256]:
```suggestion
`$SECRET` always returns a value of type string. If you want to use it in a comparison with a value that is not a string, you must perform the conversion explicitly. In the next example, the `retries` property is a number, so we need to convert the property value accordingly by calling the `tonumber` built-in jq function. 
```

**Review by kaldesai — APPROVED** [review]
The changes look good to me

**Review by gmunozfe — APPROVED** [review]
Looks good to me, perfectly done @fjtirado !

**Pull request merged**